### PR TITLE
feat(ClusterRules): Add impacted to single Cluster page

### PR DESCRIPTION
**Pull request by Fewwy**
Added the impacted column component + state when the impacted data is not available + tests

image

image

### CODE REVIEW

**Review by ikerreyes — APPROVED** [review]
Looks good to me. 

Just one note, the field is declared as optional in the API spec (https://github.com/RedHatInsights/insights-results-smart-proxy/blob/master/server/api/v2/openapi.json#L1229). I think it is an error, as it does not make sense to have a rule hit without a timestamp, but if it is possible, we should enhance the test to account for that.

I will try to clarify that.

File: src/Components/ClusterRules/ClusterRules.js [hunk 1]
```diff
@@ -244,6 +274,12 @@ const ClusterRules = ({ cluster }) => {
       sortingRows = [...rows].sort((firstItem, secondItem) => {
         const fst = firstItem[0].rule[CLUSTER_RULES_COLUMNS_KEYS[index]];
         const snd = secondItem[0].rule[CLUSTER_RULES_COLUMNS_KEYS[index]];
+        if (index === 3) {
```
Comment (gkarat) [inline src/Components/ClusterRules/ClusterRules.js:277]: please don't use hard-coded values here. Could you please map column indices with the analogy to https://github.com/RedHatInsights/ocp-advisor-frontend/blob/acba19655f0b803ee934a2c226b1489170dd2ca5/src/AppConstants.js#L264 ?

then you could use it here in a way:
```suggestion
        if (index === CLUSTER_RULES_IMPACTED_CELL) {
```

File: src/Components/ClusterRules/ClusterRules.js [hunk 2]
```diff
@@ -244,6 +274,12 @@ const ClusterRules = ({ cluster }) => {
...
+          //sorting for the impacted column
+          const firstDate = new Date(firstItem[0].rule.impacted || 0);
+          const secondDate = new Date(secondItem[0].rule.impacted || 0);
+          return firstDate > secondDate ? d : secondDate > firstDate ? -d : 0;
```
Comment (gkarat) [inline src/Components/ClusterRules/ClusterRules.js:281]: This is the same as for the line two lines below. What about declaring `fst` and `snd` with `let`, and use if-else clause with a condition `if (index === CLUSTER_RULES_IMPACTED_CELL)`, where you will then decide which values to assign to `fst` and `snd`?..

File: src/Components/ClusterRules/ClusterRules.js (hunk 1 above)
Comment (Fewwy) [reply src/Components/ClusterRules/ClusterRules.js]: Great suggestion. Thanks

File: src/Components/ClusterRules/ClusterRules.js (hunk 2 above)
Comment (Fewwy) [reply src/Components/ClusterRules/ClusterRules.js]: I applied your suggestion. Thanks

**Review by gkarat — APPROVED** [review]

**Pull request merged**